### PR TITLE
Update eslint config

### PR DESCRIPTION
**Pull request by alex00**
jobteaser https://github.com/jobteaser/jobteaser/pull/12203

### CODE REVIEW

File: rules/react.js [hunk 1]
```diff
@@ -8,38 +8,39 @@ module.exports = {
...
       {
         "skipShapeProps": true
       }
     ],
-    "react/prefer-es6-class": [ "error" ],
-    "react/prop-types": [ "warn" ],
-    "react/react-in-jsx-scope": [ "error" ],
-    "react/require-default-props": [ "error",
+    "react/prefer-es6-class": ["error"],
+    "react/prop-types": ["error"],
```
Comment (OctaveLaventure) [inline rules/react.js]: :+1:

File: rules/base.js [hunk 2]
```diff
@@ -28,47 +28,48 @@ module.exports = {
...
+    "spaced-comment": ["error", "always"],
+    "space-in-parens": ["error", "never"],
+    "space-unary-ops": ["error",
       {
         "words": true,
         "nonwords": false
       }
-    ]
+    ],
+    "prefer-destructuring": ["error", { "object": true, "array": false }]
```
Comment (OctaveLaventure) [inline rules/base.js]: :+1:

File: rules/jsx-ally.js [hunk 3]
```diff
@@ -23,5 +23,8 @@ module.exports = {
     "jsx-a11y/no-noninteractive-tabindex": [ "off" ],
     "jsx-a11y/no-static-element-interactions": [ "off" ],
     "jsx-quotes": [ "error", "prefer-double" ],
+    "jsx-a11y/click-events-have-key-events": [ "off" ],
+    "jsx-a11y/anchor-is-valid": [ "off" ],
```
Comment (Zeboch) [inline rules/jsx-ally.js]: Not sure we want to switch `off` this rule.

Comment (adrian-amigues) [reply rules/jsx-ally.js]: There were a lot of those errors, we could have kept them as warnings but we thought they would never get fixed as changing an <a> to a <button> almost always means styling updates as well. We thought that this issue should be easily caught in peer reviews from now on

Comment (Zeboch) [reply rules/jsx-ally.js]: That's why i would like to review the new rules with the teams … Here this rule sounds like a good one. But OK to go like this for now ;)

**Pull request merged**